docs/DeveloperGuide.md

```diff
@@ -1,14 +1,203 @@
 # Developer Guide
 
 ## Acknowledgements
-
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+- The `Parser` is adapted from [Dan Linh's iP](https://github.com/DanLinhHuynh-Niwashi/ip/tree/master/src/main/java/niwa/parser) code, with changes to get on well with the current project 
 
 ## Design & implementation
-
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
-
-
+### Command Parser
+The `Parser` class is responsible for interpreting user commands and extracting the associated arguments. It facilitates interaction between the user and the underlying command execution logic.
+#### Class responsibilities
+1. **Command registration**: Maintain a mapping of command words to their corresponding `Command` objects.
+2. **Command parsing**: Convert a command string entered by the user into a `Command` object.
+3. **Argument extraction**: Extract and organize the arguments associated with a given command.
+#### Class attributes
+1. **commands: LinkedHashMap<String, Command>**
+  - Description: Associates command words (as keys) with their corresponding Command objects (as values).
+#### Class main methods
+1. **registerCommands(Command command): void**
+  - **Parameters**:
+    - `command`: The `Command` object to be registered.
+  - **Process**:
+    - Retrieves the `COMMAND_WORD` field from the `Command` object
+    - Adds the word and the command to the `commands` map.
+    
+    ![register_command](./diagrams/parser/register-command-sequence.png)
+    
+2. **parseCommand(String commandPart): Command**
+  - **Parameters**: 
+    - `commandPart`: A string representing the command word entered by the user.
```

Comment:
The formatting is a bit strange on the deployed website, where I believe you expected to place `Returns:` on a new line, but it prints on the same line as the explanation for the previous part. Perhaps you could try using a double space, which is kind of the equivalent of an escape character in Markdown.